<?xml version="1.0"?>
<launch>

    <arg name="limb" default=""/>
    <arg name="control" default="joint"/> <!-- joint/cartesian -->
    <arg name="body" default="human"/> <!-- human/puppet -->

    <!-- Xsens node -->
    <!--<include file="$(find mtw_driver)/launch/demo_mtw_driver.launch"/>-->
    <!--<include file="$(find mtw_driver)/launch/mtw_driver.launch"/>-->

    <!-- Teleoperation node --> 
    <node pkg="baxter_teleop" type="baxter_mtw" name="baxter_teleop_$(arg limb)" output="screen">
      <!-- Axis and buttons parameters -->
      <rosparam file="$(find baxter_teleop)/config/mtw.yaml" command="load" />
      <!-- Twist msg name -->
      <param name="cmd_vel_topic" value="$(arg limb)/cmd_vel" type="string"/>
      <param name="cmd_pose_topic" value="$(arg limb)/cmd_pose" type="string"/>
      <param name="cmd_joint_topic" value="$(arg limb)/cmd_joint/xsens" type="string"/>
      <param name="cmd_gripper_topic" value="$(arg limb)/cmd_gripper" type="string"/>
      <param name="joint_states_topic" value="/joint_states" type="string"/>
      <param name="deadman_no_publish" value="false" type="bool"/>
      <param name="limb" value="$(arg limb)" type="string"/>
      <param name="control" value="$(arg control)" type="string"/>
      <param name="body" value="$(arg body)" type="string"/>
      <!-- Remap IMU message -->
      <remap from="/imu" to="/mtw_driver/imu/right_wrist" />
      <!-- Parameters for filters -->
      <rosparam command="load" file="$(find baxter_teleop)/launch/filter_transfer_function.yaml"/>
    </node>

</launch>

